psyneulink/library/compositions/pytorchcomponents.py

```diff
@@ -37,11 +38,49 @@ def get_fct_param_value(param_name):
         leak = get_fct_param_value('leak')
         return lambda x: (torch.max(input=(x - bias), other=torch.tensor([0], device=device).double()) * gain +
                             torch.min(input=(x - bias), other=torch.tensor([0], device=device).double()) * leak)
-
     else:
         raise Exception(f"Function {function} is not currently supported in AutodiffCompositions!")
 
-class PytorchMechanismWrapper():
+def wrap_mechanism(mechanism, index, device, context=None):
+    return PytorchMechanismWrapper(mechanism, index, device, context=context)
+
+class PytorchWrapper():
+    def _get_learnable_param_ids(self):
+        return []
+
+    def _get_learnable_param_struct_type(self, ctx):
+        structs = []
+        for param_id in self._get_learnable_param_ids():
+            param = getattr(self._object.parameters, param_id)
+            assert param is not None, f"Failed to get param {param_id} from {self}"
+            structs.append(ctx.convert_python_struct_to_llvm_ir(param.get(None)))
+        return pnlvm.ir.LiteralStructType(structs)
+
+    @property
+    def _object(self):
+        raise Exception("Unimplemented method!")
+
+    def _extract_llvm_param_ptr(self, ctx, builder, params, param_id):
+        raise Exception("Unimplemented method!")
+
+    def _copy_params_to_psyneulink(self):
+        for param_id in self._get_learnable_param_ids():
+            pytorch_param = getattr(self, param_id)
+            param = getattr(self._object.parameters, param_id)
+
+            pytorch_value = pytorch_param.detach().cpu().numpy()
+            param._set(pytorch_value, self._context)
+
+            param_port = self._object.parameter_ports[param_id]
+            param_port.parameters.value._set(pytorch_value, self._context)
```

Comment:
Why is this setting ParameterPort `value` to the same value as the base param?
Shouldn't parameter value be after modulation?

Reply:
this is dead code - removed